File: app/src/main/java/com/myown/spotat/utils/SharedPref.java
package com.myown.spotat.utils;

import android.content.Context;
import android.content.SharedPreferences;

import com.myown.spotat.R;

public class SharedPref {

    SharedPreferences pref;
    SharedPreferences.Editor editor;

    private final String DEBUG_MODE = "debugMode";
    private final String LOGIN_OTP_MODE = "loginOtpMode";

    public SharedPref(Context context) {
        pref = context.getSharedPreferences(context.getString(R.string.app_name), context.MODE_PRIVATE);
        editor= pref.edit();
    }

    public void clearAllPref() {
        editor.clear();
        editor.commit();
    }

    public void setDebugMode(boolean mode) {
        editor.putBoolean(DEBUG_MODE, mode);
        editor.commit();
    }

    public void setLoginOtpMode(boolean loginOtp) {
        editor.putBoolean(LOGIN_OTP_MODE, loginOtp);
        editor.commit();
    }

    public boolean getDebugMode() {
        return pref.getBoolean(LOGIN_OTP_MODE, false);
    }

    public boolean getLoginOtpMode() {
        return pref.getBoolean(LOGIN_OTP_MODE, false);
    }
}

File: app/src/main/java/com/myown/spotat/utils/SharedPref.kt
package com.myown.spotat.utils

import android.content.SharedPreferences
import android.content.Context
import com.myown.spotat.R

/**
 * Created by Admin on 7/30/2017.
 */
class SharedPref {

    lateinit var sharedPref: SharedPreferences
    lateinit var editor: SharedPreferences.Editor

    constructor(context : Context) {
        sharedPref = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        editor = sharedPref.edit()
    }

    val DEBUG_MODE : String = "debugMode"
    val LOGIN_OTP_MODE : String = "loginOtpMode"

    fun clearAllPref() {
        editor.clear()
        editor.commit()
    }

    fun setDebugMode(mode : Boolean) {
        editor.putBoolean(DEBUG_MODE, mode)
        editor.commit()
    }

    fun setLoginOtpMode(loginOtp : Boolean) {
        editor.putBoolean(LOGIN_OTP_MODE, loginOtp)
        editor.commit()
    }

    fun getDebugMode() : Boolean {
        return sharedPref.getBoolean(DEBUG_MODE, false)
    }

    fun getLoginOtpMode() : Boolean {
        return sharedPref.getBoolean(LOGIN_OTP_MODE, false)
    }
}
